terraform/eval_interpolate.go

```diff
@@ -31,3 +31,26 @@ func (n *EvalInterpolate) Eval(ctx EvalContext) (interface{}, error) {
 
 	return nil, nil
 }
+
+// EvalInterpolateProvider is an EvalNode implementation that takes a
+// ProviderConfig and interpolates it. Provider configurations are the only
+// "inherited" type of configuration we have, and the original raw config may
+// have a different interpolation scope.
+type EvalInterpolateProvider struct {
+	Config   *config.ProviderConfig
+	Resource *Resource
+	Output   **ResourceConfig
```

Comment:
I'm gonna need to arrange for the provider's `configschema.Block` to end up in here as part of the HCL2 work. No action required for you here... just leaving this note for myself so I can easily find this place later.